compiler/rustc_lexer/src/lib.rs

```diff
@@ -38,11 +38,11 @@ use std::convert::TryFrom;
 #[derive(Debug)]
 pub struct Token {
     pub kind: TokenKind,
-    pub len: usize,
+    pub len: u32,
```

Comment:
I bet we could use a smaller value for token, like `u16` since the minimum file size is `u32` fat32 4GiB, I wonder if this reduce the size of `Token`?

Reply:
No, because someone will need a string literal with more than 65 535 chars for example. Then u16 would overflow.

Reply:
I know, but I think even the editor cannot open it with 65535 characters in it or can it? Is it even possible to see there be a 65535 long string literal?

Reply:
Yes. It's not good style, but possible. Think about proc macros that put the content of a file into a string. 
Or a block comment can easily span a file with thousands of lines.

Somebody one day will need to hack together a way to put large binary files into the source code, and they will even hit the 4GB limit. Ideally we'd make the lengths `u64` but that is a performance tradeoff.

Reply:
Oh, I forgot comments is part of the token. Indeed comments is the one that could easily break the limit unlike the rest of the usual tokens, I was only thinking of the strings, numbers and such but not comments.